### PR TITLE
Estimation creation

### PR DESCRIPTION
**Pull request by kamleshgour**

### CODE REVIEW

File: server/models/estimationFeatureModel.js [hunk 1]
```diff
@@ -1,7 +1,18 @@
 import mongoose from 'mongoose'
 import AppError from '../AppError'
 import * as ErrorCodes from '../errorcodes'
-import {STATUS_PENDING, STATUS_APPROVED, OWNER_NEGOTIATOR, OWNER_ESTIMATOR} from "../serverconstants"
+import {validate, estimationEstimatorAddFeatureStruct} from "../validation"
+import {
```
Comment (SaurabhChouhan) [inline server/models/estimationFeatureModel.js]: Use import * as SC

File: server/models/estimationFeatureModel.js [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+    OWNER_ESTIMATOR,
+    OWNER_NEGOTIATOR,
+    STATUS_APPROVED,
+    STATUS_PENDING,
+    ROLE_ESTIMATOR,
+    ROLE_NEGOTIATOR
+} from "../serverconstants"
+import {userHasRole} from "../utils"
+import {EstimationModel, RepositoryModel} from "./"
+import {INVALID_USER, NOT_FOUND, HTTP_BAD_REQUEST} from "../errorcodes"
```
Comment (SaurabhChouhan) [inline server/models/estimationFeatureModel.js]: Use import * as EC

File: server/models/estimationFeatureModel.js [hunk 3]
```diff
@@ -42,5 +53,72 @@ let estimationFeatureSchema = mongoose.Schema({
     }]
 })
 
+
+estimationFeatureSchema.statics.addFeatureByEstimator = async (taskInput, estimator) => {
```
Comment (SaurabhChouhan) [inline server/models/estimationFeatureModel.js]: Change to featureInput

File: server/models/estimationFeatureModel.js [hunk 4]
```diff
@@ -42,5 +53,72 @@ let estimationFeatureSchema = mongoose.Schema({
...
+    }
+
+    // create estimator section
+
+    let estimatorSection = {}
+    /* Name/description would always match repository name description */
+
+    estimatorSection.name = repositoryTask.name
+    estimatorSection.description = repositoryTask.description
+    estimatorSection.estimatedHours = taskInput.estimatedHours
```
Comment (SaurabhChouhan) [inline server/models/estimationFeatureModel.js]: Estimated hours would not be sent while adding feature

File: server/models/estimationFeatureModel.js [hunk 5]
```diff
@@ -42,5 +53,72 @@ let estimationFeatureSchema = mongoose.Schema({
...
+    let estimation = await EstimationModel.findById(taskInput.estimation._id)
+    if (!estimation)
+        throw new AppError('Estimation not found', NOT_FOUND, HTTP_BAD_REQUEST)
+
+    let repositoryTask = undefined
+
+    if (taskInput.repo && taskInput.repo._id) {
+        // Feature is added from repository
+        // find if Feature actually exists there
+        repositoryTask = await RepositoryModel.findById(taskInput.repo._id)
```
Comment (SaurabhChouhan) [inline server/models/estimationFeatureModel.js]: change to repositoryFeature

File: server/models/estimationFeatureModel.js (hunk 4 above)
Comment (SaurabhChouhan) [inline server/models/estimationFeatureModel.js]: Estimator hours would not be added

File: server/validation/estimation-validation.js [hunk 6]
```diff
@@ -40,4 +40,27 @@ export const estimationEstimatorAddTaskStruct = t.struct({
...
         )
     )
+})
+
+
+export const estimationEstimatorAddFeatureStruct = t.struct({
+    _id: t.Nil,
+    name: t.maybe(RequiredString),
+    description: t.maybe(RequiredString),
+    estimatedHours: t.Number,
```
Comment (SaurabhChouhan) [inline server/validation/estimation-validation.js]: Estimated hours would not be needed while adding features

**Comment by kamleshgour** [comment]
I have updated all last requested changes

**Pull request merged**